### PR TITLE
feat!: define a single APIConfig type in the root module and referenced it from individual SDKs

### LINKED ISSUE #261: Define a single `APIConfig` type in the root module and reference it from individual SDKs.
Each submodule has the same API Config properties. It would make more sense to define a single type in the root module (in `internal`) and reference that type in a new type in each SDK.

### PR DESCRIPTION
**Pull request by alexal**
Closes #261

### CODE REVIEW

**Review by craicoverflow — CHANGES_REQUESTED** [review]
The sub modules cannot download from `github.com/redhat-developer/app-services-sdk-go/internal` because `internal` means it is not exposed. This was not an issue previously because they were not separate.

```shell
❯ cd kafkainstance
❯ go mod tidy
go: finding module for package github.com/redhat-developer/app-services-sdk-go/internal
github.com/redhat-developer/app-services-sdk-go/kafkainstance/apiv1internal imports
	github.com/redhat-developer/app-services-sdk-go/internal: module github.com/redhat-developer/app-services-sdk-go/internal: Get "https://proxy.golang.org/github.com/redhat-developer/app-services-sdk-go/internal/@v/list": dial tcp 74.125.193.141:443: i/o timeout
```

So the solution here is to put the `api_config` file into a package called `common`.

**Comment by craicoverflow** [comment]
The reason I suggested "common" was because I have seen it used in other projects ([example])(https://github.com/prometheus/client_golang/blob/master/go.mod#L9) but upon checking so is "core".

"core" makes more sense as it common as a general term for this.

**Comment by alexal** [comment]
I've moved APIConfig to core. Please review.

File: core/go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/redhat-developer/app-services-sdk-go/core
+
+go 1.13
```
Comment (craicoverflow) [inline core/go.mod:3]: "core" does not need to be its own separate module - it is part of the root module. You can just remove this file and leave everything else as is.

**Review by craicoverflow — CHANGES_REQUESTED** [review]
Once quick change needed

**Comment by alexal** [comment]
> Once quick change needed

Done.

**Review by wtrocki — APPROVED** [review]

**Review by craicoverflow — APPROVED** [review]

**Pull request merged**